src/main.rs

```diff
@@ -26,12 +31,17 @@ fn main() {
             .read_line(&mut input)
             .expect("Failed to read line");
 
+        // check if input is empty
+        if input.trim_end().len() == 0 {
+            continue;
+        }
+
         // eval
         let result = eval(&input);
 
         // print
         match result {
-            Ok(ok) => println!("{}", ok),
+            Ok(ok) => print!("{}", ok),
```

Comment:
I'm confused.  Probably ok, but what did you remove to make the newlines work?

Reply:
This change was to remove the second newline which was being output for every program. The recursion of exec_prog will always end with 'Prog::End => Ok(String::from(""))' which is output after a newline of a previous iteration of the function. Unless a program doesn't terminate in a Prog::End, we will always have a single newline at the end so we only need to print the result.

The other change to remove unnecessary newlines from the output was to use the ".trim_end()" method of a string to remove whitespace and newlines from the end.

Reply:
OK.  I think fine for now